src/lib/sitemap.test.ts

```diff
@@ -987,4 +1014,71 @@ describe('sitemap.ts', () => {
       expect(result).toEqual(expected);
     });
   });
+
+  describe('generatePathsWithRequiredlang()', () => {
+    const paths = ['/[lang]', '/[lang]/about', '/[lang]/foo/something'];
```

Comment:
- if a site has a _required_ lang param like `/[lang]`, what language is shown on its index route?  
- is it included among alternates for the root path?

Reply:
It will list each language as a separate `loc` while listing itself and the other languages as `alternate`. With the test example having this as output:

```xml
<url>
<loc>https://www.example.com/en</loc>
<xhtml:link rel="alternate" hreflang="en" href="https://www.example.com/en"/>
<xhtml:link rel="alternate" hreflang="de" href="https://www.example.com/de"/>
<xhtml:link rel="alternate" hreflang="es" href="https://www.example.com/es"/>
</url>
<url>
<loc>https://www.example.com/de</loc>
<xhtml:link rel="alternate" hreflang="en" href="https://www.example.com/en"/>
<xhtml:link rel="alternate" hreflang="de" href="https://www.example.com/de"/>
<xhtml:link rel="alternate" hreflang="es" href="https://www.example.com/es"/>
</url>
<url>
<loc>https://www.example.com/es</loc>
<xhtml:link rel="alternate" hreflang="en" href="https://www.example.com/en"/>
<xhtml:link rel="alternate" hreflang="de" href="https://www.example.com/de"/>
<xhtml:link rel="alternate" hreflang="es" href="https://www.example.com/es"/>
</url>
```

Reply:
> - if a site has a required lang param like /[lang], what language is shown on its index route?
> - is it included among alternates for the root path?

The root path is theoretically a different and unrelated page, so this has no effect. I think the root route can theoretically not exist!

> It will list each language as a separate loc while listing itself and the other languages as alternate. With the test example having this as output:

That's the behaviour I intended, going off https://developers.google.com/search/docs/specialty/international/localized-versions#sitemap.

Reply:
Got it. I'll update the readme to note that root path will not exist in the sitemap when `[lang]` is a required param and dev should redirect `/` to one of the `/xx` localized index pages.